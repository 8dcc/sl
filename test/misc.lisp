;;------------------------------------------------------------------------------
;; Miscellaneous tests that don't fit in the other files.
;;------------------------------------------------------------------------------

unbound-error-var  ; Expected: Unbound symbol: unbound-error-var

(define var1 10)         ; Expected: 10
(define var2 (* 2 var1)) ; Expected: 20

(define my-addition +)  ; Expected: <primitive ...>
(my-addition var1 var2) ; Expected: 30

;; Because there is an implicit evaluation of each argument before calling
;; `apply', `eval' receives a symbol of content "my-addition". After evaluating
;; this simbol, the addition C primitive is returned, and then applied to `var1'
;; and `var2', which were also evaluated implicitly.
(define my-symbol 'my-addition) ; Expected: my-addition
((eval my-symbol) var1 var2)    ; Expected: 30

(equal? '(1 2 3) '(1 2 3))       ; Expected: tru
(equal? '(1 2 3) '(1 2 9))       ; Expected: nil

(equal? 1 1.0) ; Expected: nil
(= 1 1.0)      ; Expected: tru
(= 1 1.000001) ; Expected: nil

;; Symbol `nil' evaluates to itself, and an empty list is converted to `nil' in
;; the parser. See the manual for more information.
(type-of nil)     ; Expected: Symbol
(type-of 'nil)    ; Expected: Symbol
(type-of ())      ; Expected: Symbol
(type-of '())     ; Expected: Symbol
(equal? nil 'nil) ; Expected: tru

(type-of '(a b c)) ; Expected: Pair
(list? '(a b c))   ; Expected: tru
(list? '(a b . c)) ; Expected: nil

(> 1 1) ; Expected: nil
(> 1 2) ; Expected: nil
(> 2 1) ; Expected: tru

(< 1 1) ; Expected: nil
(< 2 1) ; Expected: nil
(< 1 2) ; Expected: tru

(< 1.0 2) ; Expected: tru
(> 2.0 1) ; Expected: tru

(or)                       ; Expected: nil
(and)                      ; Expected: tru
(or nil nil nil)           ; Expected: nil
(and 1 2 3)                ; Expected: 3
(or nil tru 'unreachable)  ; Expected: tru
(and 123 nil 'unreachable) ; Expected: nil

(set-random-seed 50)
(random 1337)
(random 10.0)
