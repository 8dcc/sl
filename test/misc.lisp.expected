Error: Unbound symbol: `unbound-error-var'.
10
20
(1 2 3)
foo
(1 foo 3)
<primitive 0xDEADBEEF>
30
my-addition
30
tru
nil
nil
tru
nil
Symbol
Symbol
Symbol
Symbol
tru
Pair
tru
nil
nil
nil
tru
nil
nil
tru
tru
tru
nil
tru
nil
3
tru
nil
tru
1195
6.286430
